speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/InterventionConfig.cpp

```diff
@@ -6,12 +6,29 @@
 
 #include "InterventionConfig.h"
 
+void InitializeSounds(std::unordered_map<InterventionAction, const char*>& p_sounds, void* p_xmlHandle, unsigned int p_interventionCount) {
+    char path[256];
+    for (int i = 0; i < p_interventionCount; i++)
+    {
+        snprintf(path, sizeof(path), "%s/%d/%s", PRM_SECT_INTERVENTIONS, i, PRM_SECT_SOUND);
+        if(!GfParmExistsSection(p_xmlHandle, path)) continue;
+
+        const char* src = GfParmGetStr(p_xmlHandle, path, PRM_ATTR_SRC, "");
+        char* soundPath = new char[256];
+        snprintf(soundPath, 256, SOUNDS_DIR_FORMAT, GfDataDir(), src);
+
+        p_sounds[i] = soundPath;
+    }
+}
+
 void InterventionConfig::Initialize() {
```

Comment:
It is better to place the documentation at the implementation of the function (in the cpp file instead of the header).